### PR TITLE
Added Menu Hours for Individual Meals of Day

### PR DESCRIPTION
**Pull request by kaushikravikumar**
### Summary 

This Pull Request adds the operating hours text for each individual meal of day, as the user navigates between them. It also addresses a bug that I spotted. Occasionally, the Menu API response had an empty menu for a meal of day. For example, Rose opens up for breakfast, but had an empty menu for breakfast. In this case, there was 3 operating hour time intervals, but only 2 menus. So now the code addresses that by still adding an empty menuItem to the menu object for the breakfast meal. This way, the tab for breakfast shows up, as it should! 

121253955_440317206940625_4245785203383145766_n

### Test Plan 

Go through different dining hall facility info pages and verify that the hours match up with those displayed on the Cornell Eateries website. As per my tests, it seems to match up perfectly!

### CODE REVIEW

File: Density/app/src/main/java/org/cornelldti/density/density/network/API.kt [hunk 1]
```diff
@@ -12,6 +12,7 @@ import org.cornelldti.density.density.data.MenuClass
 import org.cornelldti.density.density.data.OperatingHoursClass
 import org.json.JSONArray
 import org.json.JSONException
+import org.json.JSONObject
```
Comment (kungpaogao) [inline Density/app/src/main/java/org/cornelldti/density/density/network/API.kt:15]: was this by accident?

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt [hunk 2]
```diff
@@ -276,9 +271,12 @@ class FacilityInfoPage : BaseActivity() {
...
                     setOpenOrClosedOnToolbar()
                 },
-                facilityHoursStringsOnResponse = {
+
+                facilityHoursStringsOnResponse = { opHoursStringsList ->
+                    opHoursStrings = opHoursStringsList
                     fetchMenuJSON(FluxUtil.getCurrentDate(), facilityClass!!.id)
-                })
+                }
+        )
```
Comment (kungpaogao) [inline Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:279]: this doesn't work - we only call `fetchOperatingHours` once, so it won't change when you change the days

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt [hunk 3]
```diff
@@ -493,6 +491,16 @@ class FacilityInfoPage : BaseActivity() {
...
 
+            // This is where the operating hours for the selected meal of day is set!
+            if(availableMenus.isNotEmpty() && opHoursStrings.isNotEmpty()) {
+                menuHours.text = opHoursStrings[availableMenus.indexOf(mealOfDay)]
+                clock_image.visibility = View.VISIBLE
+            }
+            else {
+                menuHours.text = ""
+                clock_image.visibility = View.INVISIBLE
+            }
```
Comment (kungpaogao) [inline Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:502]: I haven't been able to test the edge cases, but you almost always want to get visibility to `View.GONE` so it doesn't take up layout space

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt [hunk 4]
```diff
@@ -493,6 +491,16 @@ class FacilityInfoPage : BaseActivity() {
                 adapter = menuItemListViewAdapter
             }
 
+            // This is where the operating hours for the selected meal of day is set!
+            if(availableMenus.isNotEmpty() && opHoursStrings.isNotEmpty()) {
+                menuHours.text = opHoursStrings[availableMenus.indexOf(mealOfDay)]
+                clock_image.visibility = View.VISIBLE
```
Comment (kungpaogao) [inline Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:497]: This doesn't work because we don't fetch operating hours more than once (which might be wrong to begin with, now that I think about it).

**Review by kungpaogao — CHANGES_REQUESTED** [review]
You should be using the menu endpoint (`fetchMenuJSON`) to fetch operating hours because it provides the start and end times for each meal. This way you don't need to try to fit data from two different endpoints. 

In terms of implementation, this will probably require you to rewrite the menu data model so it includes a start and end time for each meal.

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt (hunk 4 above)
Comment (kungpaogao) [reply Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:497]: This currently causes an out of bounds error.

**Comment by kaushikravikumar** [comment]
> You should be using the menu endpoint (`fetchMenuJSON`) to fetch operating hours because it provides the start and end times for each meal. This way you don't need to try to fit data from two different endpoints.
> 
> In terms of implementation, this will probably require you to rewrite the menu data model so it includes a start and end time for each meal.

that's fair!! Was thinking the same thing as I was implementing lol

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt (hunk 2 above)
Comment (kaushikravikumar) [reply Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:279]: I'm pretty sure we've always wanted the top menu to remain the same tho, right? Like that should always show today's open/closed data

Comment (kaushikravikumar) [reply Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:279]: ohh yes good catch

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt (hunk 4 above)
Comment (kaushikravikumar) [reply Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:497]: wait what about it does? i never got an out of bounds error

Comment (kungpaogao) [reply Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt]: yup, try something that doesn't have the same meals everyday like Becker; also, you can tell based on the code because the hours don't refresh

Comment (kaushikravikumar) [reply Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt]: > This doesn't work because we don't fetch operating hours more than once (which might be wrong to begin with, now that I think about it).

The way I've implemented it now is that it will fetch operating hours once using the operating hours endpoint. This fetch gets the current day's hours and the first time slot of tomorrow so that it can accurately display the open/closed and when its open until or next opens on the toolbar. Then there is also a fetch for operating hours everytime the menu api endpoint is hit. The operating hours are a part of the menu object now, so in the showMenu() function, it now just displays the corresponding open hours for a specific meal of day of the currently selected menu.

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt [hunk 5]
```diff
@@ -92,9 +83,6 @@ class FacilityInfoPage : BaseActivity() {
         topBar.title = facilityClass!!.name
         topBar.setNavigationOnClickListener { onBackPressed() }
 
-        feedback = findViewById(R.id.accuracy)
-        feedback.movementMethod = LinkMovementMethod.getInstance()
```
Comment (bungaepyo) [inline Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:96]: Is there a reason to remove accuracy rn? Aren't we planning to implement the feedback feature within a few weeks?

**Review by bungaepyo — COMMENTED** [review]
Except for the comment, LGTM!

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt (hunk 5 above)
Comment (kaushikravikumar) [reply Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:96]: yeaah but the current feedback system is kinda useless. Like we're gonna change it up entirely, so I figured we'd add it back once we had a new system. But keeping the old google form link just felt like it would be taking up space in the app.

Comment (bungaepyo) [reply Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:96]: Okay gotcha!

**Review by bungaepyo — APPROVED** [review]

**Review by kungpaogao — APPROVED** [review]
Looks good
